app/sign.js

```diff
@@ -276,13 +277,45 @@ const handleSignErc20 = function(recoveryRequest, key, skipConfirm) {
   return transaction.serialize().toString('hex');
 };
 
+/* *
+  Takes in either an xprv, xlmsecret, or 24 words.
+  Returns an xprv or xlmsecret
+ */
+const parseKey = function(rawkey, coin, path) {
+
+  if(rawkey.includes(',') && rawkey.split(',').length === 24) {
+    const mnemonic = rawkey.replace(/,/g,' '); // replace commas with spaces
+    if(coin === 'xlm' || coin === 'txlm') {
+      // stellar is special (thanks Jeb)
+      const stellarWallet = stellarHd.fromMnemonic(mnemonic);
+      return stellarWallet.getSecret(0);
+    }
+
+    // every other coin can use xpubs
+    if(!bip39.validateMnemonic(mnemonic)) {
+      throw new Error("Invalid mnemonic");
+    }
+    const seed = bip39.mnemonicToSeed(mnemonic);
+    let node = utxoLib.HDNode.fromSeedBuffer(seed);
+    if(path) {
+      node = node.derivePath(path);
+    }
+    const xprv = node.toBase58();
+    return xprv;
+
+  }
+  // if it doesn't have commas, we expect it is an xprv or xlmsecret properly formatted
+  return key;
+}
+
 const handleSign = function(args) {
   const file = args.file;
-  const key = args.key;
 
   const recoveryRequest = JSON.parse(fs.readFileSync(file, { encoding: 'utf8' }));
   const coin = recoveryRequest.coin;
 
+  const key = parseKey(args.key, coin, args.path);
```

Comment:
We should prompt for this if it isn't provided in the arguments.

Reply:
updated